Tests/Microsoft.AppCenter.Test.Windows/Storage/StorageTest.cs

```diff
@@ -459,18 +459,70 @@ public async Task SaveLogPurgesOldestLogsWhenStorageIsFull()
             Assert.IsTrue(setMaxStorageSizeResult);
 
             // Fill databse with logs.
-            var initialLogs = await FillDatabaseWithLogs(capacity);
+            await FillDatabaseWithLogs(capacity);
+
+            // Get first log
+            var initialLogs = new List<Log>();
+            await _storage.GetLogsAsync(StorageTestChannelName, int.MaxValue, initialLogs);
             var firstLog = initialLogs[0];
+            var secondLog = initialLogs[1];
 
             // Create a new log. Vverify that new log was added and old was deleted.
             var newLog = TestLog.CreateTestLog();
             await _storage.PutLog(StorageTestChannelName, newLog);
             var retrievedLogs = new List<Log>();
+            await _storage.ClearPendingLogState(StorageTestChannelName);
             await _storage.GetLogsAsync(StorageTestChannelName, int.MaxValue, retrievedLogs);
             CollectionAssert.Contains(retrievedLogs, newLog);
+            CollectionAssert.Contains(retrievedLogs, secondLog);
             CollectionAssert.DoesNotContain(retrievedLogs, firstLog);
         }
 
+        /// <summary>
+        /// Verify that Storage doesn't loop infinitely if old logs cannot be purged.
+        /// Verify that Storage doesn't save a new log in case it cannot purge older ones.
+        /// </summary>
+        [TestMethod]
+        public async Task SaveLogDoesntPurgesOldLogsWhenStorageIsFullOfDifferentKindOfLogs()
+        {
+            // Prepare constants
+            var anotherChannelName = "anotherChannelName";
+
+            // Set storage max size.
+            var capacity = 12 * 1024;
+            var setMaxStorageSizeResult = await _storage.SetMaxStorageSizeAsync(capacity);
+            Assert.IsTrue(setMaxStorageSizeResult);
+
+            // Fill databse with logs.
+            await FillDatabaseWithLogs(capacity);
+
+            // Get stored collection
+            var initialLogs = new List<Log>();
+            await _storage.GetLogsAsync(StorageTestChannelName, int.MaxValue, initialLogs);
+
+            // Try to add a new log with another channel name to storage.
+            // Vverify that StorageException is thrown.
+            var newLog = TestLog.CreateTestLog();
+            try
+            {
+                await _storage.PutLog(anotherChannelName, newLog);
+            }
```

Comment:
Let's crash the test here if the error wasn't called (like [here](https://github.com/microsoft/appcenter-sdk-dotnet/blob/3ea78d13475ad1f9a71b0ed9cf5ab8fbff767636/Tests/Microsoft.AppCenter.Test.Windows/AppCenterTest.cs#L681))